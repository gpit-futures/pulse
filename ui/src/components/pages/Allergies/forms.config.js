export const valuesNames = {
  CAUSE: 'cause',
  REACTION: 'reaction',
  CAUSECODE: 'causeCode',
  TERMINOLOGY: 'causeTerminology',
  TERMINOLOGYCODE: 'terminologyCode',
  AUTHOR: 'author',
  DATE: 'currentDate',
  DATE_CREATED: 'dateCreated',

  IMPORT: 'originalSource',
  ISIMPORT: 'isImport',
  ORIGINAL_SOURCE: 'originalSource',
  ORIGINAL_COMPOSITION: 'originalComposition',

  SOURCE: 'source',
  SOURCE_ID: 'sourceId',
};

export const valuesLabels = {
  CAUSE: 'Cause',
  REACTION: 'Reaction',
  TERMINOLOGY: 'Terminology',
  CAUSECODE: 'Cause Code',
  TERMINOLOGYCODE: 'Terminology Code',
  AUTHOR: 'Author',
  DATE: 'Date',
  DATE_CREATED: 'Date',

  IMPORT: 'Import URL',
  ISIMPORT: 'Is Imported',
  ORIGINAL_SOURCE: 'Original Source',

  SOURCE: 'Source',
  SOURCE_ID: 'Source ID',
};
