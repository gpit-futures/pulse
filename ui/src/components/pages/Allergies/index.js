import { combineEpics } from 'redux-observable';

import asyncComponent from '../../../components/containers/AsyncComponent/AsyncComponent';
import { clientUrls } from '../../../config/client-urls.constants';

import { fetchPatientAllergiesDetailEpic } from './ducks/fetch-patient-allergies-detail.duck';
import { fetchPatientAllergiesDetailEditEpic } from './ducks/fetch-patient-allergies-detail-edit.duck';
import { fetchPatientAllergiesEpic } from './ducks/fetch-patient-allergies.duck';
import { fetchPatientAllergiesUpdateEpic } from './ducks/fetch-patient-allergies.duck';
import { fetchPatientAllergiesCreateEpic } from './ducks/fetch-patient-allergies-create.duck';

import patientsAllergies from './ducks/fetch-patient-allergies.duck';
import patientAllergiesCreate from './ducks/fetch-patient-allergies-create.duck';
import allergiesDetail from './ducks/fetch-patient-allergies-detail.duck';
import allergiesDetailEdit from './ducks/fetch-patient-allergies-detail-edit.duck';

const epics = combineEpics(fetchPatientAllergiesDetailEpic, fetchPatientAllergiesDetailEditEpic, fetchPatientAllergiesEpic, fetchPatientAllergiesCreateEpic, fetchPatientAllergiesUpdateEpic);
const Allergies = asyncComponent(() => import(/* webpackChunkName: "allergies" */ './Allergies').then(module => module.default));

const reducers = {
  patientsAllergies,
  patientAllergiesCreate,
  allergiesDetail,
  allergiesDetailEdit,
};

const sidebarConfig = { key: 'allergies', pathToTransition: '/allergies', name: 'Allergies', isVisible: true };

const routers = [
  { key: 'allergies', component: Allergies, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.ALLERGIES}` },
  { key: 'allergiesCreate', component: Allergies, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.ALLERGIES}/create` },
  { key: 'allergiesDetail', component: Allergies, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.ALLERGIES}/:sourceId` },
];

export default {
  component: Allergies,
  epics, reducers, sidebarConfig, routers,
}
