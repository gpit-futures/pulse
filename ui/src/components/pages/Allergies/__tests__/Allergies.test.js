import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import Allergies from '../Allergies';
import { valuesNames } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

// frequently used variables
const userId = '9999999000';
const sourceId = 'acb0eaf2-d1df-4c7a-9382-619b31935f2b';

// storage setup in different configurations
const mockStore = configureStore();
const storeResource = {
  patientsAllergies: {
    '9999999024': [
      {
        [valuesNames.SOURCE_ID]: sourceId,
        [valuesNames.SOURCE]: 'EtherCIS',
        text: 'fsfsfs',
      },
    ],
  },
};
const storeWithFormsError = mockStore(Object.assign({
  allergiesDetail: {
    '9999999000': {
      [valuesNames.CAUSE]: 111,
      [valuesNames.CAUSECODE]: 111,
      [valuesNames.TERMINOLOGY]: 1239085,
      [valuesNames.TERMINOLOGYCODE]: 111,
      [valuesNames.REACTION]: 111,
      [valuesNames.AUTHOR]: 'Dr Tony Shannon',
      [valuesNames.DATE_CREATED]: 1507019519000,
      [valuesNames.SOURCE]: 'ethercis',
      [valuesNames.SOURCE_ID]: sourceId,
      [valuesNames.ORIGINAL_COMPOSITION]: '',
      [valuesNames.ORIGINAL_SOURCE]: '',
    },
  },
  form: {
    allergiePanelFormSelector: {
      syncErrors: {
        [valuesNames.CAUSE]: 'You must enter a value.',
        [valuesNames.REACTION]: 'You must enter a value.',
      },
    },
    allergiesCreateFormSelector: {
      syncErrors: {
        [valuesNames.CAUSE]: 'You must enter a value.',
        [valuesNames.REACTION]: 'You must enter a value.',
      },
    },
  },
}, storeResource));
const storeEmpty = mockStore(Object.assign({
  allergiesDetail: {
    '9999999000': {},
  },
}, storeResource));
const storeWithDetail = mockStore(Object.assign({
  allergiesDetail: {
    '9999999000': {
      [valuesNames.CAUSE]: 111,
      [valuesNames.CAUSECODE]: 111,
      [valuesNames.TERMINOLOGY]: 1239085,
      [valuesNames.TERMINOLOGYCODE]: 111,
      [valuesNames.REACTION]: 111,
      [valuesNames.AUTHOR]: 'Dr Tony Shannon',
      [valuesNames.DATE_CREATED]: 1507019519000,
      [valuesNames.SOURCE]: 'ethercis',
      [valuesNames.SOURCE_ID]: sourceId,
      [valuesNames.ORIGINAL_COMPOSITION]: '',
      [valuesNames.ORIGINAL_SOURCE]: '',
    },
  },
}, storeResource));

// configure context for various tests
const generateNewContext = (oldContext, pathname) => {
  const newContext = Object.assign({}, oldContext);
  newContext.router = Object.assign({}, newContext.router);
  newContext.router.history = Object.assign({}, newContext.router.history);
  newContext.router.history.location = { pathname };
  return newContext;
};
const context = {
  router: {
    history: {
      push: () => {},
      replace: () => {},
      location: {
        pathname: `/patients/${userId}/allergies`,
      },
    },
    route: {
      match: {
        params: {
          sourceId,
          userId,
        },
      },
    },
  },
};
const contextCreate = generateNewContext(context, `/patients/${userId}/allergies/create`);
const contextDetail = generateNewContext(context, `/patients/${userId}/allergies/${sourceId}`);

const contextImport = {
  router: {
    route: { match: { params: { userId } } },
    history: {
      push: () => {},
      replace: () => {},
      goBack: () => {},
      location: {
        pathname: `/patients/${userId}/allergies/create`,
        state: {
          importData: {
            isImport: true,
            originalSource: 'domen.com/documents/documents_id',
            cause: 'cause',
          }
        }
      },
    }
  },
};

// configuration of forms for testing methods
const formValuesEdit = {
  [valuesNames.CAUSE]: 112,
  [valuesNames.REACTION]: 112,
  [valuesNames.AUTHOR]: 'Dr Tony Shannon',
};
const formValuesCreate = {
  [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
  [valuesNames.CAUSECODE]: '1239085',
  [valuesNames.SOURCE_ID]: '',
  [valuesNames.ISIMPORT]: false,
  [valuesNames.TERMINOLOGYCODE]: '12393890',
  [valuesNames.TERMINOLOGY]: 'SNOMED-CT',
  [valuesNames.CAUSE]: 'test',
  [valuesNames.REACTION]: 'test',
};
const formValuesImportCreate = {
  [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
  [valuesNames.CAUSECODE]: '1239085',
  [valuesNames.SOURCE_ID]: '',
  [valuesNames.ISIMPORT]: true,
  [valuesNames.IMPORT]: 'domen.com/documents/documents_id',
  [valuesNames.TERMINOLOGYCODE]: '12393890',
  [valuesNames.TERMINOLOGY]: 'SNOMED-CT',
  [valuesNames.CAUSE]: 'cause',
  [valuesNames.REACTION]: 'test',
};
const match = {
  params: {
    userId,
  },
};

describe('Component <Allergies />', () => {
  it('should renders correctly with allergiesDetail and testing Detail Panel', () => {
    const component = shallow(
      <Allergies
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive().dive();

    // Testing component handleDetailAllergiesClick methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('AllergiesDetail')).toHaveLength(0);
    expect(component.find('PluginCreate')).toHaveLength(0);

    component.instance().handleDetailAllergiesClick('065d85e3-3cd5-4604-bb94-5685fffb193d');
    const componentStateAfterMethod = component.state();
    component.setState({ isSecondPanel: true, isDetailPanelVisible: true, isBtnExpandVisible: true, isBtnCreateVisible: true, isCreatePanelVisible: false, openedPanel: 'allergiePanel', editedPanel: {}, isLoading: true, expandedPanel: 'all' });
    const componentStateAfterSetState = component.state();

    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('AllergiesDetail')).toHaveLength(1);
    expect(component.find('PluginCreate')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    // Testing component handleShow methods
    expect(component.state().openedPanel).toEqual('allergiePanel');
    component.instance().handleShow('metaPanel');
    expect(component.state().openedPanel).toEqual('metaPanel');
    component.instance().handleShow('allergiePanel');
    expect(component.state().openedPanel).toEqual('allergiePanel');

    // Testing component handleExpand methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.state().openedPanel).toEqual('allergiePanel');
    expect(component.state().expandedPanel).toEqual('all');
    component.instance().handleExpand('allergiePanel', 'allergiesDetail');
    expect(component.state().openedPanel).toEqual('allergiePanel');
    expect(component.state().expandedPanel).toEqual('allergiePanel');
    component.setState({ openedPanel: 'allergiePanel', expandedPanel: 'allergiePanel' });
    expect(component.find('PluginListHeader')).toHaveLength(0);
    expect(component.find('PluginMainPanel')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    component.setState({ openedPanel: 'allergiePanel', expandedPanel: 'all' });

    component.instance().handleExpand('allergiePanel', 'allergiesCreate');
    component.setState({ expandedPanel: 'all' });
    component.instance().handleExpand('allergiePanel', 'allergiesMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('allergiePanel', 'allergiesMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('allergiePanel', 'allergiesCreate');

    // Testing component detail form methods
    expect(component.state().editedPanel).toEqual({});
    component.instance().handleEdit('allergiePanel');
    expect(component.state().editedPanel).toEqual({ allergiePanel: true});
    component.instance().handleAllergieDetailCancel('allergiePanel');
    expect(component.state().editedPanel).toEqual({ allergiePanel: false});
    component.instance().handleEdit('allergiePanel');
    component.instance().handleSaveSettingsDetailForm(formValuesEdit, 'allergiePanel');
    expect(component.state().editedPanel).toEqual({ allergiePanel: false});
    component.instance().handleSaveSettingsDetailForm(formValuesEdit, 'metaPanel');

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with allergiesDetail and testing Create Panel', () => {
    const component = shallow(
      <Allergies
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive().dive();

    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('AllergiesDetail')).toHaveLength(0);
    expect(component.find('PluginCreate')).toHaveLength(0);

    // Testing component create panel methods
    component.instance().handleCreate();
    const componentStateAfterMethod = component.state();
    component.setState({ isBtnCreateVisible: false, isCreatePanelVisible: true, openedPanel: 'allergiesCreate', isSecondPanel: true, isDetailPanelVisible: false, isLoading: true, isBtnExpandVisible: true, expandedPanel: 'all', isSubmit: false })
    const componentStateAfterSetState = component.state();
    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('PluginCreate')).toHaveLength(1);
    expect(component).toMatchSnapshot();
    component.instance().handleCreateCancel();
    component.instance().handleSaveSettingsCreateForm(formValuesCreate);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing another methods', () => {
    const component = shallow(
      <Allergies
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive().dive();

    // Testing component hideCreateForm methods
    component.instance().hideCreateForm();
    expect(component.state().isBtnCreateVisible).toEqual(true);
    expect(component.state().isCreatePanelVisible).toEqual(false);
    expect(component.state().openedPanel).toEqual('allergiePanel');
    expect(component.state().isSecondPanel).toEqual(false);
    expect(component.state().expandedPanel).toEqual('all');
    expect(component.state().isBtnExpandVisible).toEqual(false);

    // Testing component handleFilterChange methods
    expect(component.state().nameShouldInclude).toEqual('');
    component.instance().handleFilterChange({ target: { value: 'test' } });
    expect(component.state().nameShouldInclude).toEqual('test');
    component.instance().handleFilterChange({ target: { value: '' } });
    expect(component.state().nameShouldInclude).toEqual('');

    // Testing component handleHeaderCellClick methods
    expect(component.state().columnNameSortBy).toEqual('cause');
    expect(component.state().sortingOrder).toEqual('asc');
    component.instance().handleHeaderCellClick({}, { name: 'reaction', sortingOrder: 'desc' });
    expect(component.state().columnNameSortBy).toEqual('reaction');
    expect(component.state().sortingOrder).toEqual('desc');

    // Testing component handleSetOffset methods
    expect(component.state().offset).toEqual(0);
    component.instance().handleSetOffset(10);
    expect(component.state().offset).toEqual(10);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing context and lifecycle componentWillReceiveProps', () => {
    const component = shallow(
      <Allergies
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive().dive();

    component.setProps({ test: 'testing context' });
    component.setContext(contextCreate);
    component.setProps({ test: 'testing create context' });
    component.setContext(contextDetail);
    component.setProps({ test: 'testing edit context' });

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with forms error', () => {
    const component = shallow(
      <Allergies
        store={storeWithFormsError}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive().dive();

    component.instance().handleSaveSettingsDetailForm(formValuesEdit, 'allergiePanel');
    component.instance().handleSaveSettingsCreateForm(formValuesCreate);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with empty store', () => {
    const component = shallow(
      <Allergies
        store={storeEmpty}
        match={match}
      />, { context }).dive().dive().dive().dive().dive().dive().dive();

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly when data take from Documents how "import"', () => {
    const component = shallow(
      <Allergies
        store={storeEmpty}
        match={match}
      />, { context: contextImport }).dive().dive().dive().dive().dive().dive().dive();

    component.instance().goBack();
    component.instance().handleSaveSettingsCreateForm(formValuesImportCreate);

    expect(component).toMatchSnapshot();
  });
});
