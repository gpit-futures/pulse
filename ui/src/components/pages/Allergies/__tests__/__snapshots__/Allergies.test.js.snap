// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Allergies /> should renders correctly and testing another methods 1`] = `
<section
  className="page-wrapper"
>
  <PluginBanner
    img="allergies.jpg"
    subTitle="Short blurb containing a few words to describe this section"
    title="Allergies"
    toRight={false}
  />
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="allergiesMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="allergiesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Allergies"
          />
          <PluginMainPanel
            columnNameSortBy="cause"
            emptyDataMessage="No allergies"
            headers={
              Array [
                Object {
                  "key": "cause",
                  "title": "Cause",
                  "width": "25%",
                },
                Object {
                  "key": "reaction",
                  "title": "Reaction",
                  "width": "55%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "Source ID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Allergies /> should renders correctly and testing context and lifecycle componentWillReceiveProps 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="allergiesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="allergiesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Allergies"
          />
          <PluginMainPanel
            columnNameSortBy="cause"
            emptyDataMessage="No allergies"
            headers={
              Array [
                Object {
                  "key": "cause",
                  "title": "Cause",
                  "width": "25%",
                },
                Object {
                  "key": "reaction",
                  "title": "Reaction",
                  "width": "55%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "Source ID",
                  "width": 0,
                },
              ]
            }
            id="acb0eaf2-d1df-4c7a-9382-619b31935f2b"
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <AllergiesDetail
          currentPanel="allergiesDetail"
          detail={
            Object {
              "author": "Dr Tony Shannon",
              "cause": 112,
              "causeCode": 111,
              "causeTerminology": 1239085,
              "dateCreated": 1507019519000,
              "originalComposition": "",
              "originalSource": "",
              "reaction": 112,
              "source": "ethercis",
              "sourceId": "acb0eaf2-d1df-4c7a-9382-619b31935f2b",
              "terminologyCode": 111,
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          isSubmit={false}
          name="allergiesDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="allergiesCreate"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Allergies /> should renders correctly when data take from Documents how "import" 1`] = `
<section
  className="page-wrapper"
>
  <PluginBanner
    img="allergies.jpg"
    subTitle="Short blurb containing a few words to describe this section"
    title="Allergies"
    toRight={false}
  />
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="allergiesMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="allergiesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Allergies"
          />
          <PluginMainPanel
            columnNameSortBy="cause"
            emptyDataMessage="No allergies"
            headers={
              Array [
                Object {
                  "key": "cause",
                  "title": "Cause",
                  "width": "25%",
                },
                Object {
                  "key": "reaction",
                  "title": "Reaction",
                  "width": "55%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "Source ID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Allergies /> should renders correctly with allergiesDetail and testing Create Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="allergiesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="allergiesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Allergies"
          />
          <PluginMainPanel
            columnNameSortBy="cause"
            emptyDataMessage="No allergies"
            headers={
              Array [
                Object {
                  "key": "cause",
                  "title": "Cause",
                  "width": "25%",
                },
                Object {
                  "key": "reaction",
                  "title": "Reaction",
                  "width": "55%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "Source ID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <PluginCreate
          componentForm={
            <ReduxForm
              isSubmit={false}
            />
          }
          currentPanel="allergiesCreate"
          expandedPanel="all"
          isCreatePanelVisible={true}
          isImport={false}
          name="allergiesCreate"
          onCancel={[Function]}
          onExpand={[Function]}
          onGoBack={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="allergiesCreate"
          title="Create Allergy"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Allergies /> should renders correctly with allergiesDetail and testing Create Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="allergiesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="allergiesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Allergies"
          />
          <PluginMainPanel
            columnNameSortBy="cause"
            emptyDataMessage="No allergies"
            headers={
              Array [
                Object {
                  "key": "cause",
                  "title": "Cause",
                  "width": "25%",
                },
                Object {
                  "key": "reaction",
                  "title": "Reaction",
                  "width": "55%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "Source ID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <PluginCreate
          componentForm={
            <ReduxForm
              isSubmit={false}
            />
          }
          currentPanel="allergiesCreate"
          expandedPanel="all"
          isCreatePanelVisible={true}
          isImport={false}
          name="allergiesCreate"
          onCancel={[Function]}
          onExpand={[Function]}
          onGoBack={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="allergiesCreate"
          title="Create Allergy"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Allergies /> should renders correctly with allergiesDetail and testing Detail Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="allergiesMain"
            isBtnExpandVisible={true}
            isBtnTableVisible={false}
            name="allergiesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Allergies"
          />
          <PluginMainPanel
            columnNameSortBy="cause"
            emptyDataMessage="No allergies"
            headers={
              Array [
                Object {
                  "key": "cause",
                  "title": "Cause",
                  "width": "25%",
                },
                Object {
                  "key": "reaction",
                  "title": "Reaction",
                  "width": "55%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "Source ID",
                  "width": 0,
                },
              ]
            }
            id="acb0eaf2-d1df-4c7a-9382-619b31935f2b"
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <AllergiesDetail
          currentPanel="allergiesDetail"
          detail={
            Object {
              "author": "Dr Tony Shannon",
              "cause": 111,
              "causeCode": 111,
              "causeTerminology": 1239085,
              "dateCreated": 1507019519000,
              "originalComposition": "",
              "originalSource": "",
              "reaction": 111,
              "source": "ethercis",
              "sourceId": "acb0eaf2-d1df-4c7a-9382-619b31935f2b",
              "terminologyCode": 111,
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          isSubmit={false}
          name="allergiesDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="allergiePanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Allergies /> should renders correctly with allergiesDetail and testing Detail Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section full-panel full-panel-details"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <AllergiesDetail
          currentPanel="allergiesDetail"
          detail={
            Object {
              "author": "Dr Tony Shannon",
              "cause": 111,
              "causeCode": 111,
              "causeTerminology": 1239085,
              "dateCreated": 1507019519000,
              "originalComposition": "",
              "originalSource": "",
              "reaction": 111,
              "source": "ethercis",
              "sourceId": "acb0eaf2-d1df-4c7a-9382-619b31935f2b",
              "terminologyCode": 111,
            }
          }
          editedPanel={Object {}}
          expandedPanel="allergiePanel"
          isSubmit={false}
          name="allergiesDetail"
          onCancel={[Function]}
          onEdit={[Function]}
          onExpand={[Function]}
          onSaveSettings={[Function]}
          onShow={[Function]}
          openedPanel="allergiePanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Allergies /> should renders correctly with allergiesDetail and testing Detail Panel 3`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    />
  </div>
</section>
`;

exports[`Component <Allergies /> should renders correctly with empty store 1`] = `
<section
  className="page-wrapper"
>
  <PluginBanner
    img="allergies.jpg"
    subTitle="Short blurb containing a few words to describe this section"
    title="Allergies"
    toRight={false}
  />
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="allergiesMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="allergiesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Allergies"
          />
          <PluginMainPanel
            columnNameSortBy="cause"
            emptyDataMessage="No allergies"
            headers={
              Array [
                Object {
                  "key": "cause",
                  "title": "Cause",
                  "width": "25%",
                },
                Object {
                  "key": "reaction",
                  "title": "Reaction",
                  "width": "55%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "Source ID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Allergies /> should renders correctly with forms error 1`] = `
<section
  className="page-wrapper"
>
  <PluginBanner
    img="allergies.jpg"
    subTitle="Short blurb containing a few words to describe this section"
    title="Allergies"
    toRight={false}
  />
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="allergiesMain"
            isBtnExpandVisible={false}
            isBtnTableVisible={false}
            name="allergiesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Allergies"
          />
          <PluginMainPanel
            columnNameSortBy="cause"
            emptyDataMessage="No allergies"
            headers={
              Array [
                Object {
                  "key": "cause",
                  "title": "Cause",
                  "width": "25%",
                },
                Object {
                  "key": "reaction",
                  "title": "Reaction",
                  "width": "55%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "20%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "Source ID",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={true}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onCreate={[Function]}
            onHeaderCellClick={[Function]}
            setOffset={[Function]}
            sortingOrder="asc"
            totalEntriesAmount={0}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;
